swap_meet/vendor.py

```diff
@@ -1,2 +1,95 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__ (self, inventory = None):
+        if not inventory:
+            inventory = []
+            
+        self.inventory = inventory
+
+    def add(self, item): 
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        try: 
+            self.inventory.remove(item)
+            return item
+
+        except:
+            return False
+
+    def get_by_category(self, category):
+        items = []
+
+        for item in self.inventory:
+            if item.category == category:
+                items.append(item)
+
+        return items
+
+    def swap_items(self, vendor, my_item, their_item):
+        '''
+        add my item to vendor's inventory
+        remove my item from my inventory
+        add their item to my inventory
+        remove their item from their inventory
+        ---
+        I initially used the above add() and remove() methods, but wanted something a bit more logically streamlined. 
+        Accomplishing the swap of items using a tuple allows for a direct in-place substitution without iterating over the lists.
+        However, I'm not sure if list index() and list remove() have significantly different time complexity - both seem to be O(n).
+        In addition, using this method throws an error if either item is not in the list, 
+        which allows me to return False using try-except, instead of manually checking and returning False.
+        '''
+        try:
+            my_item_index = self.inventory.index(my_item)
+            their_item_index = vendor.inventory.index(their_item)
+
+            swap_items = self.inventory[my_item_index], vendor.inventory[their_item_index]
+            vendor.inventory[their_item_index], self.inventory[my_item_index] = swap_items
```

Comment:
Great job on this streamlined approach. The `add` and `remove` methods are now just taking up space. I understand that you technically need them to pass the tests but just wanted to make that note.